packages/test/test-end-to-end-tests/src/test/summarizerWithLocalChanges.spec.ts

```diff
@@ -110,57 +166,191 @@ async function createSummarizer(
 		container,
 		rootDataObjectFactory,
 		summaryVersion,
-		undefined,
+		undefined /* containerRuntimeFactoryType */,
 		registryStoreEntries,
+		undefined /* logger */,
+		mockConfigProvider(settings),
 	);
 }
 
-describeNoCompat(
-	"Data store realized between startSummary and summarize",
-	(getTestObjectProvider) => {
-		let provider: ITestObjectProvider;
+async function waitForSummaryOp(container: IContainer): Promise<boolean> {
+	return new Promise<boolean>((resolve) => {
+		container.deltaManager.on("op", (op: ISequencedDocumentMessage) => {
+			if (op.type === MessageType.Summarize) {
+				resolve(true);
+			}
+		});
+	});
+}
 
-		const createContainer = async (): Promise<IContainer> => {
-			return provider.createContainer(runtimeFactory);
-		};
+describeNoCompat("Summarizer with local data stores", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
 
-		async function waitForSummary(summarizer: ISummarizer) {
-			const summaryResult = await summarizeNow(summarizer);
-			return summaryResult;
-		}
+	beforeEach(async () => {
+		provider = getTestObjectProvider({ syncSummarizer: true });
+		settings = [];
+		settings["Fluid.ContainerRuntime.Test.SummaryStateUpdateMethod"] = "restart";
+		settings["Fluid.ContainerRuntime.Test.CloseSummarizerDelayOverrideMs"] = 0;
+	});
 
-		beforeEach(async () => {
-			provider = getTestObjectProvider({ syncSummarizer: true });
-		});
+	itExpects(
+		"summary should fail before generate stage when data store is created during summarize",
+		[
+			{
+				eventName: "fluid:telemetry:Summarizer:Running:Summarize_cancel",
+				clientType: "noninteractive/summarizer",
+				error: "NodeDidNotRunGC",
+			},
+		],
+		async () => {
+			const container = await createContainer(provider);
+			await waitForContainerConnection(container);
+			const rootDataObject = await requestFluidObject<RootTestDataObject>(container, "/");
+			const dataObject = await dataStoreFactory1.createInstance(
+				rootDataObject.containerRuntime,
+			);
+			rootDataObject._root.set("dataStore2", dataObject.handle);
+			const { summarizer } = await createSummarizer(provider, container);
+			await provider.ensureSynchronized();
+
+			// Summarization should fail because of a data store created during summarization which does not run GC.
+			await assert.rejects(
+				async () => {
+					await summarizeNow(summarizer);
+				},
```

Comment:
nit/FYI - you could do this (same as replacing `await` with `return`, and then more concise)
```suggestion
				async () => summarizeNow(summarizer),
```

Reply:
Good tip.